app/api/helpers/calendar/ical.py

```diff
@@ -39,10 +39,12 @@ def to_ical(event, include_sessions=False, my_schedule=False):
             .order_by(Session.starts_at.asc())
         )
         if my_schedule:
-            if not current_user:
-                return jsonify(error='Login Required'), 401
+            if not (current_user or user_id):
+                return jsonify(error='Forbidden Access'), 401
+
+            user_id = current_user.id if not user_id else user_id
```

Comment:
```suggestion
            user_id = user_id or current_user.id
```

Reply:
Ohh I need to start writing better code :joy: